src/test/programs/federal/snap/unit.test.js

```diff
@@ -535,13 +543,257 @@ describe('SNAPhelpers', () => {
+    });
+
+    it('that has a negative income after deductions, it should return 0', () => {
+      adjustedGross = 0;
+      standardDeduction = earnedIncomeDeduction = medicalDeduction = dependentCareDeduction = 1;
+      mockHelpers()
+      
+      const deductions = standardDeduction + earnedIncomeDeduction + medicalDeduction + dependentCareDeduction;
+      const income = adjustedGross - deductions;
+      expect(income).toBeLessThan(0)
+      expect(SNAPhelpers.getAdjustedNotGrossIncome( current )).toEqual(0);
+    });
+  });
+
+
   // `SNAPhelpers.monthlyNetIncome()`
+  describe('`.monthlyNetIncome( timeClient )` given a time-restricted client object with', () => {
+    let current, mocks, income, homelessDeduction, shelterDeduction;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      mocks = [];
+    });
+    afterEach(() => {
+      mocks.forEach(mock => mock.mockRestore());
+    });
+
+    const mockHelpers = () => {
+      mocks = [
+        jest.spyOn(SNAPhelpers, 'getAdjustedNotGrossIncome').mockReturnValue(income),
+        jest.spyOn(SNAPhelpers, 'getHomelessDeduction').mockReturnValue(homelessDeduction),
+        jest.spyOn(SNAPhelpers, 'getShelterDeduction').mockReturnValue(shelterDeduction)
+      ];
+    };
+
+    it('returns a adjusted income less certain deductions', () => {
+      income = 1000;
+      homelessDeduction = 100;
+      shelterDeduction = 1;
+      mockHelpers()
+      
+      const monthlyNetIncome = income - homelessDeduction - shelterDeduction;
+      expect(SNAPhelpers.monthlyNetIncome( current )).toEqual(monthlyNetIncome);
+    });
+
+    it('that has a negative income after deductions, it should return 0', () => {
+      income = 0;
+      homelessDeduction = 100;
+      shelterDeduction = 1;
+      mockHelpers()
+      
+      const monthlyNetIncome = income - homelessDeduction - shelterDeduction;
+      expect(monthlyNetIncome).toBeLessThan(0);
+      expect(SNAPhelpers.monthlyNetIncome( current )).toEqual(0);
+    });
+  })
+
+
   // `SNAPhelpers.getMaxNetIncome()`
+  describe('`.getMaxNetIncome( timeClient )` given a time-restricted client object with', () => {
+    let current, getPovertyGrossIncomeLevel;
+    beforeEach(() => {
+      current = cloneDeep( defaultCurrent );
+      getPovertyGrossIncomeLevel = jest.spyOn(SNAPhelpers, 'getPovertyGrossIncomeLevel');
+    });
+    afterEach(() => {
+      getPovertyGrossIncomeLevel.mockRestore();
+    })
+
+    it('that has income below the poverty line, it should return \'no limit\'', () => {
+      getPovertyGrossIncomeLevel.mockReturnValue(Number.POSITIVE_INFINITY);
+      expect(SNAPhelpers.getMaxNetIncome( current )).toEqual('no limit');
+    });
+
+    it('that has no elderly or disabled members, it should return \'no limit\'', () => {
+      getPovertyGrossIncomeLevel.mockReturnValue(Number.NEGATIVE_INFINITY);
+      expect(SNAPhelpers.hasDisabledOrElderlyMember( current )).toBe(false);
```

Comment:
Maybe this function should be mocked too?. Maybe all three functions it uses should be mocked? I'm not sure how far we take this mocking stuff.